src/script/page/RightSidebar/MessageDetails/MessageDetails.tsx

```diff
@@ -0,0 +1,276 @@
+import {UserRepository} from '../../../user/UserRepository';
+import {formatLocale} from 'Util/TimeUtil';
+import {isContentMessage} from '../../../guards/Message';
+import {Message} from '../../../entity/message/Message';
+import {SearchRepository} from '../../../search/SearchRepository';
+import {UserReactionMap} from '../../../storage';
+
+const MESSAGE_STATES = {
+  LIKES: 'likes',
+  NO_LIKES: 'no-likes',
+  NO_RECEIPTS: 'no-receipts',
+  RECEIPTS: 'receipts',
+  RECEIPTS_OFF: 'receipts-off',
+};
+
+const formatUserCount = (users: User[]): string => (users.length ? ` (${users.length})` : '');
+
+const sortUsers = (userA: User, userB: User): number =>
+  userA.name().localeCompare(userB.name(), undefined, {sensitivity: 'base'});
+
+const formatTime = (time: string | number | Date) => formatLocale(time, 'P, p');
+
+interface MessageDetailsProps {
+  activeConversation: Conversation;
+  onClose: () => void;
+  conversationRepository: ConversationRepository;
+  messageEntity: Message;
+  teamRepository: TeamRepository;
+  searchRepository: SearchRepository;
+  userRepository: UserRepository;
+  showLikes?: boolean;
+  updateEntity: (message: Message) => void;
+}
+
+const MessageDetails: FC<MessageDetailsProps> = ({
+  activeConversation,
+  conversationRepository,
+  messageEntity,
+  teamRepository,
+  searchRepository,
+  showLikes = false,
+  userRepository,
+  onClose,
+  updateEntity,
+}) => {
+  const [receiptUsers, setReceiptUsers] = useState<User[]>([]);
+  const [likeUsers, setLikeUsers] = useState<User[]>([]);
+  const [messageId, setMessageId] = useState<string>(messageEntity.id);
+
+  const [isReceiptsOpen, setIsReceiptsOpen] = useState<boolean>(showLikes);
+
+  const {timestamp, user: selfUser} = useKoSubscribableChildren(messageEntity, ['timestamp', 'user']);
+
+  const teamId = activeConversation.team_id;
+  const supportsReceipts = selfUser.isMe && teamId;
+
+  const supportsLikes = useMemo(() => {
+    const isPing = messageEntity.super_type === SuperType.PING;
+    const isEphemeral = messageEntity?.isEphemeral();
+
+    return !isPing && !isEphemeral;
+  }, [messageEntity]);
+
+  const messageState = useMemo(() => {
+    if (supportsReceipts && isReceiptsOpen) {
+      if (!messageEntity.expectsReadConfirmation) {
+        return MESSAGE_STATES.RECEIPTS_OFF;
+      }
+
+      return receiptUsers.length ? MESSAGE_STATES.RECEIPTS : MESSAGE_STATES.NO_RECEIPTS;
+    }
+
+    return likeUsers.length ? MESSAGE_STATES.LIKES : MESSAGE_STATES.NO_LIKES;
+  }, [supportsReceipts, isReceiptsOpen, messageEntity, receiptUsers, likeUsers]);
+
+  const getLikes = useCallback(async (reactions: UserReactionMap) => {
+    const currentLikes = Object.keys(reactions);
+    const usersLikes = await userRepository.getUsersById(currentLikes.map(likeId => ({domain: '', id: likeId})));
+    const sortedUsersLikes = usersLikes.sort(sortUsers);
+
+    setLikeUsers(sortedUsersLikes);
+  }, []);
+
+  const receipts = messageEntity?.readReceipts() || [];
```

Comment:
This will need to be watched (as it can be live updated) you will need to use `useKoSub` hook